### PR TITLE
added real-path resolver for addon properties

### PR DESCRIPTION
**Pull request by elgatito**
ADDON_PATH is resolved by real script location.
Chain: 
- get real script location
- decode into utf-8, depending on filesystemencoding (yes, that is another one, related to files and their paths)
- get it's directory
- browse 3 levels under

That path does not need to be encoded/decoded when opening files, as long as the script has a "utf-8" header on the first line.

Tests on windows with cyrillic profile. Daemon started, I can browse Quasar, start a play, video is played.
When I stop if - Quasar crashes. - http://paste.ubuntu.com/24058516/

So need to adjust the daemon, python should work fine.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Translation

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by elgatito** [comment]
It's a fix for #547

**Comment by scakemyer** [comment]
That crash doesn't seem related to addon_path at all to be honest... Can you try to restart Kodi, start fresh, and check if it's reproducible?

**Comment by elgatito** [comment]
I will re-read provided logs. My environment does not have this bug.

File: resources/site-packages/quasar/addon.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
+# -*- coding: utf-8 -*-
+import os
+import sys
 import xbmcaddon
 
 ADDON = xbmcaddon.Addon()
 ADDON_ID = ADDON.getAddonInfo("id")
 ADDON_NAME = ADDON.getAddonInfo("name")
-ADDON_PATH = ADDON.getAddonInfo("path").decode('utf-8')
+ADDON_PATH = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(__file__).decode(sys.getfilesystemencoding(), 'ignore')), "../../../"))
```
Comment (scakemyer) [inline resources/site-packages/quasar/addon.py]: Should be `"..", "..", ".."` instead at the end

File: resources/site-packages/quasar/rpc.py [hunk 2]
```diff
@@ -139,6 +139,7 @@ def GetAddonInfo(self):
                     "fanart", "icon", "id", "name", "path", "profile", "stars",
                     "summary", "type", "version"):
             info[key] = ADDON.getAddonInfo(key)
+            info['path'] = ADDON_PATH
```
Comment (scakemyer) [inline resources/site-packages/quasar/rpc.py]: Might as well remove `path` from the keys we get.

Comment (scakemyer) [inline resources/site-packages/quasar/rpc.py]: Also, don't set it for every key, just once after they've all been set ;)

**Comment by elgatito** [comment]
Okay. So after a review of logs and output:
- Daemon queries RPC for addon information, which is proxied from xbmcaddon(), so it's up to Kodi which is there.
``` 
panic: html/template: pattern matches no files: `C:\Users\XBMC\AppData\Roaming\Kodi\addons\plugin.video.quasar\resources\web\*.html`
``` 
This line is directly showing that ADDON.path is totally wrong. (why - who knows).
So we use detector in addon.py to get real path of an addon, and give it to Daemon.

I can't test since i'm on Windows 7 and Windows Store required 10 to install.

**Comment by elgatito** [comment]
Tabs, tabs .. 

I have added a fallback setter (legacy Kodi's settings path-through). It is possible getfilesystemencoding() returns bullish value that can't be decoded, so then we pass legacy value.

File: resources/site-packages/quasar/addon.py [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
 import xbmcaddon
 
 ADDON = xbmcaddon.Addon()
 ADDON_ID = ADDON.getAddonInfo("id")
 ADDON_NAME = ADDON.getAddonInfo("name")
-ADDON_PATH = ADDON.getAddonInfo("path").decode('utf-8')
 ADDON_ICON = ADDON.getAddonInfo("icon").decode('utf-8')
+try:
+    ADDON_PATH = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(__file__).decode(sys.getfilesystemencoding(), 'ignore')), "..", "..", ".."))
+except Exception as e:
```
Comment (scakemyer) [inline resources/site-packages/quasar/addon.py]: No need for `Exception as e`

**Comment by scakemyer** [comment]
Good, I like that.

**Comment by elgatito** [comment]
Last change up here. Would be good to test it.

**Comment by scakemyer** [comment]
Looking great :+1:

**Pull request merged**